From 8f691206e4aa769cc99e264f1c15ae6956789f91 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@strace.io>
Date: Mon, 15 May 2023 08:00:00 +0000
Subject: [PATCH] configure: do not take the SUBLEVEL part of the kernel
 version into account

* configure.ac: When deciding whether to use bundled version of kernel
headers, ignore the SUBLEVEL part of the kernel version, assuming that
any potential changes in UAPI introduced in stable kernels are not
important.

Resolves: https://github.com/strace/strace/issues/253
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index efd0632e80..f12ad7a725 100644
--- a/configure.ac
+++ b/configure.ac
@@ -278,7 +278,7 @@ AC_CACHE_CHECK(
 	[AS_IF([test $enable_bundled = check],
 	       [AC_COMPILE_IFELSE(
 			[AC_LANG_PROGRAM([#include <linux/version.h>
-#define USE_OS_HEADERS (LINUX_VERSION_CODE > ]linux_version_code[ ? 1 : -1)],
+#define USE_OS_HEADERS ((LINUX_VERSION_CODE >> 8) > (]linux_version_code[ >> 8) ? 1 : -1)],
 					 [[int i[USE_OS_HEADERS];]]
 			 )
 			],
